code/player/Player.RPCs.cs

```diff
@@ -34,6 +36,7 @@ public void ClientSetRole(string roleName)
         public void ClientConfirmPlayer(TTTPlayer confirmPlayer, TTTPlayer deadPlayer, string roleName)
         {
             deadPlayer.Role = RoleFunctions.GetRoleByType(RoleFunctions.GetRoleTypeByName(roleName));
+            deadPlayer.IsAlive = false;
```

Comment:
Instead of `IsAlive` it should be maybe `IsConfirmedDead`, `IsAlive` makes it sounds like it's the state of the player, whereas this seems to be confirming a death.

Reply:
That is a good way to additionally keep track of the confirmation state on the server. Feel free modifying the check, currently you can't set IsAlive on server because there is LifeState available (and should be used) :)